src/scripting/lua_ui.cc

```diff
@@ -326,9 +330,493 @@ int LuaPanel::indicate(lua_State* L) {
+
+             * ``"raised"`` (default)
+             * ``"permpressed"``
+             * ``"flat"``
+
+           * ``"repeating"``: **Optional**. Whether pressing and holding
+             the button generates repeated events. Default: false.
+
+           * ``"on_click"``: **Optional**. Callback code to run when the
+             button is clicked. To associate actions with a button press,
+             prefer this over the ``on_panel_clicked`` event.
+
+      Note that event callbacks functions must be provided as raw code in string form.
+      During the lifetime of a *toolbar* widget, the Lua Interface used by the game may be reset.
+      Therefore, any callbacks attached to such widgets must not use any functions or variables
+      defined at an arbitrary earlier time by your script -
+      they may have been deleted by the time the callback is invoked.
+      Example:
+
+      .. code-block:: lua
+
+         push_textdomain("yes_no.wad", true)
+
+         local mv = wl.ui.MapView()
+
+         mv.toolbar:create_child({
+            widget   = "button",
+            name     = "yes_no_window",
+            w        = styles.get_size("toolbar_button_size"),
+            h        = styles.get_size("toolbar_button_size"),
+            tooltip  = _("Yes or No"),
+            icon     = "images/ui_basic/different.png",
+            on_click = [[
+               push_textdomain("yes_no.wad", true)
+               wl.ui.MapView():create_child({
+                  widget   = "window",
+                  name     = "yes_or_no_window",
+                  title    = _("Yes or No"),
+                  x        = wl.ui.MapView().width  // 2,
+                  y        = wl.ui.MapView().height // 2,
+                  content  = {
+                     widget      = "box",
+                     orientation = "vert",
+                     children    = {
+                        {
+                           widget = "textarea",
+                           font   = "wui_info_panel_paragraph",
+                           text   = _("Click Yes or No"),
+                        },
+                        {
+                           widget = "space",
+                           value  = 10,
+                        },
+                        {
+                           widget   = "button",
+                           name     = "yes",
+                           title    = _("Yes"),
+                           on_click = [=[
+                              wl.ui.show_messagebox(_("Hello"), _("You clicked yes!"))
+                           ]=],
+                        },
+                        {
+                           widget   = "button",
+                           name     = "no",
+                           title    = _("No"),
+                           on_click = [=[
+                              wl.ui.show_messagebox(_("Foo"), _("You clicked no!"), false)
+                           ]=],
+                        },
```

Comment:
-- The buttons are one below each other
-- If you want them side by side you need to enclose them in another widget of type box with orientation horizontal